### PR TITLE
Clean up some of the docs

### PR DESCRIPTION
**Pull request by RubenRBS**
These are some minor issues that I found while fixing a few of the ones that @ErniGH reported when reading the tutorial

### CODE REVIEW

File: tutorial/versioning/lockfiles.rst [hunk 1]
```diff
@@ -315,9 +315,12 @@ configuration, and then merge them:
 This multiple-clean + merge operation is not something that developers should do, only CI
 scripts, and for some advanced CI flows that will be explained later.
 
+.. note:
+
+   It is possible to lock down to package revisions, but this would be not recommended for
+   most use cases, and should only be used in extreme and problematic cases.
+
```
Comment (memsharded) [inline tutorial/versioning/lockfiles.rst:322]:
```suggestion
```

We have even removed this from the command help and from the docs, it is not mentioned anywhere.  So it is not really possible at the moment.

File: tutorial/consuming_packages/the_flexibility_of_conanfile_py.rst [hunk 2]
```diff
@@ -190,9 +190,12 @@ platform without adding more changes.
 
         def requirements(self):
             self.requires("zlib/1.2.11")
+            if self.settings.os == "Windows":
```
Comment (memsharded) [inline tutorial/consuming_packages/the_flexibility_of_conanfile_py.rst:193]: This needs to be reviewed by @czoido, cause this tutorial has always been a bit challenging to follow in sync the code and the docs

**Review by memsharded — COMMENTED** [review]
Great job

File: tutorial/consuming_packages/the_flexibility_of_conanfile_py.rst (hunk 2 above)
Comment (RubenRBS) [reply tutorial/consuming_packages/the_flexibility_of_conanfile_py.rst:193]: This is in fact an update to the docs to sync them back :)

**Review by czoido — APPROVED** [review]

**Pull request merged**